vite.config.js

```diff
@@ -1,55 +1,63 @@
-import { defineConfig } from 'vite'
+import { defineConfig, loadEnv } from 'vite'
 import react from '@vitejs/plugin-react'
 import path from 'path';
-import getEnvModule from './env'
 import dns from 'dns'
 
-dns.setDefaultResultOrder('verbatim')
+// https://vitejs.dev/config/
+export default defineConfig(({ _, mode }) => {
 
-getEnvModule().setEnvironmentVarsFromDir(__dirname);
+  const env = loadEnv(mode, process.cwd(), '')
 
-process.env.CLIENT_ID = process.env.SPA_CLIENT_ID || process.env.CLIENT_ID;
+  env.CLIENT_ID = env.SPA_CLIENT_ID || env.CLIENT_ID
 
-const env = {};
+  const customEnv = {}
 
-// List of environment variables made available to the app
-[
-  'ISSUER',
-  'CLIENT_ID',
-  'EXT_AUTH_URL',
-  'EXT_CLIENT_ID',
-  'EXT_REDIRECT_URI',
-  'LINK_API',
-  'SUPPORT_EMAIL',
-  'SUPPORT_URL',
-].forEach((key) => {
-  if (!process.env[key]) {
-    throw new Error(`Environment variable ${key} must be set. See README.md`);
-  }
-  env[key] = process.env[key];
-});
+  const variables = [
+      'ISSUER',
+      'CLIENT_ID',
+      'EXT_AUTH_URL',
+      'EXT_CLIENT_ID',
+      'EXT_REDIRECT_URI',
+      'LINK_API',
+      'SUPPORT_EMAIL',
+      'SUPPORT_URL'
+  ]
 
-// https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [react()],
-  define: {
-    'process.env': env
-  },
-  resolve: {
-    alias: {
-      'react-router-dom': path.resolve(__dirname, 'node_modules/react-router-dom')
+  variables.forEach((key) => {
+    if (!env[key]) {
+      throw new Error(`Environment variable ${key} must be set`)
     }
-  },
-  server: {
-    port: process.env.PORT || 3000
-  },
-  build: {
-    rollupOptions: {
-      // always throw with build warnings
-      onwarn (warning, warn) {
-        warn('\nBuild warning happened, customize "onwarn" callback in vite.config.js to handle this error.');
-        //throw new Error(warning);
+    customEnv[key] = env[key]
+  })
+
+  dns.setDefaultResultOrder('verbatim')
```

Comment:
Can you see any reason why our Okta PS consultant added this? It appears superfluous since it is setting the default option and should only make a difference if IPv6 is misconfigured. https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder

I'd prefer we remove this and
```
import dns from 'dns'
```
if they are not needed, as long as we are cleaning up this file.

Reply:
Looking into this...

Reply:
Looking at the Vite docs, this option is to configure the server that runs Vite and serves the React app to the user.
Seems that there are cases that if the configured domain to server the app resolves to multiple IPs (like localhost), Vite might not serve the request if there is a mismatch between Browser and server. This dns fix is used to address it. From the docs: https://vite.dev/config/server-options

I tested without and it works as before. Not sure it was happening to the Okta PS consultant.

We can remove and test the results in an environment to see if is necessary

Reply:
This is now removed